colossalai/inference/struct.py

```diff
@@ -197,16 +208,30 @@ def init_batch(self, seqs: List["Sequence"] = None):
 
                 self.sequences_set.add(seq)
 
+    def init_fd_tensors(self):
```

Comment:
This function seems to be redundant as it is called multiple times to try to initialize the singleton. You might want to just initialize the singleton in cache manager, and then just left `fd_inter_tensor: FDIntermTensors = FDIntermTensors()` to point to the singleton.

Reply:
Done